core/services/workflows/v2/engine_test.go

```diff
@@ -215,3 +218,61 @@ func newTriggerSubs(n int) *wasmpb.ExecutionResult {
 		},
 	}
 }
+
+func TestEngine_Execution(t *testing.T) {
+	t.Parallel()
+
+	module := modulemocks.NewModuleV2(t)
+	module.EXPECT().Start()
+	module.EXPECT().SetCapabilityExecutor(mock.Anything).Return(nil)
+	module.EXPECT().Close()
+	capreg := regmocks.NewCapabilitiesRegistry(t)
+	capreg.EXPECT().LocalNode(matches.AnyContext).Return(capabilities.Node{}, nil)
+
+	initDoneCh := make(chan error)
+	subscribedToTriggersCh := make(chan []string, 1)
+	executionFinishedCh := make(chan string)
+
+	cfg := defaultTestConfig(t)
+	cfg.Module = module
+	cfg.CapRegistry = capreg
+	cfg.Hooks = v2.LifecycleHooks{
+		OnInitialized: func(err error) {
+			initDoneCh <- err
+		},
+		OnSubscribedToTriggers: func(triggerIDs []string) {
+			subscribedToTriggersCh <- triggerIDs
+		},
+		OnExecutionFinished: func(executionID string) {
+			executionFinishedCh <- executionID
+		},
+	}
+
+	t.Run("successful execution with no capability calls", func(t *testing.T) {
+		engine, err := v2.NewEngine(t.Context(), cfg)
+		require.NoError(t, err)
+		module.EXPECT().Execute(matches.AnyContext, mock.Anything).Return(newTriggerSubs(1), nil).Once()
+		trigger := capmocks.NewTriggerCapability(t)
+		capreg.EXPECT().GetTrigger(matches.AnyContext, "id_0").Return(trigger, nil).Once()
+		eventCh := make(chan capabilities.TriggerResponse)
+		trigger.EXPECT().RegisterTrigger(matches.AnyContext, mock.Anything).Return(eventCh, nil).Once()
+		trigger.EXPECT().UnregisterTrigger(matches.AnyContext, mock.Anything).Return(nil).Once()
+
+		require.NoError(t, engine.Start(t.Context()))
+
+		require.NoError(t, <-initDoneCh) // successful trigger registration
+		require.Equal(t, []string{"id_0"}, <-subscribedToTriggersCh)
+
+		module.EXPECT().Execute(matches.AnyContext, mock.Anything).Return(nil, nil).Once()
+		eventCh <- capabilities.TriggerResponse{
+			Event: capabilities.TriggerEvent{
+				TriggerType: "basic-trigger@1.0.0",
+				ID:          "event_012345",
+				Payload:     nil,
+			},
+		}
+		<-executionFinishedCh
+
+		require.NoError(t, engine.Close())
+	})
+}
```

Comment:
super nit, could you add a second test that runs the test executable used to test the module? It'll ensure that there's no gaps to re-use it. I wouldn't remove this test, since it's easier to debug if things go wrong.

I don't use config in that test yet, but it probably should opt to soon.

Reply:
This test is blocked until `SetCapabilityExecutor` is implemented on the host.  So I've split this test out to a separate [ticket](https://smartcontract-it.atlassian.net/browse/CAPPL-844).